### DISCUSSION TITLE
quad instead of triangular elements in 2D-surface mesh

### DISCUSSION THREAD

**Discussion by BenBeans** [nschloe/pygmsh#523] [Q&A] [answered]
I'm trying to build a 2D-surface mesh using the add_polygon feature. It works fine but I can just generate a triangulated mesh. is there an argument I can pass to geom.generate_mesh() for example to generate a quadratic mesh? Thanks in advance!

**Comment by nschloe** [comment, marked as answer]
`generate_mesh(order=2)` gives you quadratic triangular elements.

**Reply by BenBeans** [reply]
I think I caused a misunderstanding. I would like to have quadrilateral elements, no triangles.

**Reply by nschloe** [reply]
I see. The keyword you're looking for is `recombine`, which combines two triangles to a quad. Check out [this example](https://github.com/nschloe/pygmsh/blob/main/tests/built_in/test_quads.py).

**Reply by BenBeans** [reply]
it works like a charm... thanks. Please keep up your great work!
